extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -56,64 +56,88 @@ require([
 			}
 		};
 
-	if (window.wgContentLanguage !== 'en') {
-		if (videosModule) {
-			videosModule('#recirculation-rail');
-		}
-		oldDiscussions();
-		return;
-	}
+	function prepareEnglishRecirculation () {
+		// prepare & render right rail recirculation module
+		liftigniter.prepare(railRecirculation).done(function (data) {
+			require(['ext.wikia.recirculation.views.premiumRail'], function (viewFactory) {
+				var view = viewFactory();
+				view.render(data)
+					.then(view.setupTracking())
+					.then(function () {
+						liftigniter.setupTracking(view.itemsSelector, railRecirculation);
+					});
+			});
+		});
 
-	// prepare & render right rail recirculation module
-	liftigniter.prepare(railRecirculation).done(function (data) {
-		require(['ext.wikia.recirculation.views.premiumRail'], function (viewFactory) {
-			var view = viewFactory();
-			view.render(data)
-				.then(view.setupTracking())
-				.then(function () {
-					liftigniter.setupTracking(view.itemsSelector, railRecirculation);
+		// prepare & render mixed content footer module
+		var mixedContentFooterData = [
+			liftigniter.prepare(mixedContentFooter.nsItems),
+			liftigniter.prepare(mixedContentFooter.wikiItems),
+			discussions.prepare()
+		];
+		$.when.apply($, mixedContentFooterData).done(function (nsItems, wikiItems, discussions) {
+			$mixedContentFooterContent.show();
+			require(['ext.wikia.recirculation.views.mixedFooter'], function (viewFactory) {
+				var view = viewFactory();
+				view.render({
+					nsItems: nsItems,
+					wikiItems: wikiItems,
+					discussions: discussions
+				}).then(function () {
+					liftigniter.setupTracking(view.nsItemsSelector, mixedContentFooter.nsItems);
+					liftigniter.setupTracking(view.wikiItemsSelector, mixedContentFooter.wikiItems);
 				});
+			});
 		});
-	});
+	}
 
-	// prepare & render mixed content footer module
-	var mixedContentFooterData = [
-		liftigniter.prepare(mixedContentFooter.nsItems),
-		liftigniter.prepare(mixedContentFooter.wikiItems),
-		discussions.prepare()
-	];
-	$.when.apply($, mixedContentFooterData).done(function (nsItems, wikiItems, discussions) {
-		$mixedContentFooterContent.show();
-		require(['ext.wikia.recirculation.views.mixedFooter'], function (viewFactory) {
-			var view = viewFactory();
-			view.render({
-				nsItems: nsItems,
-				wikiItems: wikiItems,
-				discussions: discussions
-			}).then(function () {
-				liftigniter.setupTracking(view.nsItemsSelector, mixedContentFooter.nsItems);
-				liftigniter.setupTracking(view.wikiItemsSelector, mixedContentFooter.wikiItems);
+	function prepareInternationalRecirculation () {
+		var mixedContentFooterData = [
+			liftigniter.prepare(mixedContentFooter.wikiItems),
+			discussions.prepare()
+		];
+		$.when.apply($, mixedContentFooterData).done(function (wikiItems, discussions) {
+			$mixedContentFooterContent.show();
```

Comment:
This looks pretty similar to https://github.com/Wikia/app/pull/13670/files#diff-c7077e2816c4f2e1464b8533b6fcc3cfR79 -> Could we create a reusable method that takes a callback and render params?